### PR TITLE
Add Kaldi

### PR DESCRIPTION
**Pull request by mcernak**
closes #2618

### CODE REVIEW

**Comment by mcernak** [comment]
@conda-forge/staged-recipes Can I ask you for a review of that recipe? GCC 4.8.1 and 4.8.2 have a bug in the implementation of the nth_element algorithm provided by the standard library. This will cause Kaldi to crash. Therefore I specify the GCC version in the meta file.

**Comment by ericdill** [comment]
This is more than I'm comfortable tackling. Maybe @jakirkham, @ocefpaf or @scopatz could jump in on this one?

File: recipes/kaldi/build.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#! /usr/bin/env bash
+
+pushd src
+
+./configure --shared --mathlib=OPENBLAS --openblas-root=${PREFIX} --fst-root=${PREFIX} --fst-version=1.6.1 --speex-root=${PREFIX} --use-cuda=no
+make -j ${CPU_COUNT}
```
Comment (ocefpaf) [inline recipes/kaldi/build.sh]: Is there a `make check` we can run here?

File: recipes/kaldi/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package:
+  name: kaldi
+  version: r7271.1a4dbf6
+
+source:
+  git_url: https://github.com/kaldi-asr/kaldi.git
+  git_rev: 1a4dbf6be118e4967cf4efd510811ef72cdfee3d
+  patches:
+    - conda-kaldi.patch
```
Comment (ocefpaf) [inline recipes/kaldi/meta.yaml]: Can you add a comment describing the patch? What is it for and/or if it was submitted upstream?

File: recipes/kaldi/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package:
+  name: kaldi
+  version: r7271.1a4dbf6
+
+source:
+  git_url: https://github.com/kaldi-asr/kaldi.git
```
Comment (ocefpaf) [inline recipes/kaldi/meta.yaml]: Please use the tarball format. See https://github.com/conda-forge/nco-feedstock/blob/master/recipe/meta.yaml#L8-L10 for an example.

**Comment by ocefpaf** [comment]
Looks almost good to go. Just a dew minor comments/clarifications and we can merge this.

File: recipes/kaldi/build.sh (hunk 1 above)
Comment (mcernak) [inline recipes/kaldi/build.sh]: Unfortunately there is no make check.

File: recipes/kaldi/meta.yaml [hunk 4]
```diff
@@ -3,7 +3,8 @@ package:
   version: r7271.1a4dbf6
 
 source:
-  git_url: https://github.com/kaldi-asr/kaldi.git
+  fn: master.tar.gz
+  url: https://github.com/kaldi-asr/kaldi/archive/master.tar.gz
   git_rev: 1a4dbf6be118e4967cf4efd510811ef72cdfee3d
```
Comment (ocefpaf) [inline recipes/kaldi/meta.yaml]: Your case should be:

```yaml
{% set version = "r7271.1a4dbf6" %}
{% set commit = "1a4dbf6be118e4967cf4efd510811ef72cdfee3d" %}

 package:
   name: kaldi
    version: {{ version }}
  
  source:
    fn: {{ version }}-{{ commit }}.tar.gz
    url: https://github.com/kaldi-asr/kaldi/archive/{{ commit }}.tar.gz
    sha256: 33d7ebbfb66f2271521bc51b49dc93dd5ec1704d9469aafe46e07b6e51de5d92
```

**Comment by mcernak** [comment]
@ocefpaf Thank you. Kaldi builds in place, so the patch makes just compilation of the libraries in ${PREFIX}. The patch also adds sources for two binaries that could be moved to upstream (it is not submitted there yet).

**Comment by ocefpaf** [comment]
> @ocefpaf Thank you. Kaldi builds in place, so the patch makes just compilation of the libraries in ${PREFIX}. The patch also adds sources for two binaries that could be moved to upstream (it is not submitted there yet).

Just add a comment in the `meta.yaml` right above the patch about that. It is meant to help others understand the patch easily.

**Comment by mcernak** [comment]
@ocefpaf Patch comment added.

**Pull request merged**